internal/provider/baremetal/installConfig.go

```diff
@@ -40,7 +41,18 @@ func (p baremetalProvider) AddPlatformToInstallConfig(
 			p.Log.Warnf("Failed to unmarshall Host %s inventory", hostutil.GetHostnameForMsg(host))
 			return err
 		}
-		hosts[yamlHostIdx].BootMACAddress = inventory.Interfaces[0].MacAddress
+
+		for _, iface := range inventory.Interfaces {
+			if iface.MacAddress != "" {
+				hosts[yamlHostIdx].BootMACAddress = iface.MacAddress
+			}
+		}
+		if hosts[yamlHostIdx].BootMACAddress == "" {
```

Comment:
I'm not sure if this can happen, after all the agent was able to connect to the service somehow no?

Reply:
Reading the code it looks like something we should check for even if we don't expect it to happen in reality. I'm fine with it.

Reply:
sounds reasonable